internal/cypress/docker/runner.go

```diff
@@ -71,6 +75,52 @@ func (r *Runner) RunProject() (int, error) {
 	return errorCount, nil
 }
 
+// checkCypressVersionAvailability do several checks before running Cypress tests.
+func (r *Runner) checkCypressVersionAvailability() error {
+	// Skip availability check since custom image is being used
+	if r.Project.Docker.Image.Name != "" && r.Project.Docker.Image.Tag != "" {
+		log.Info().Msgf("Ignoring Cypress version for Docker, using %s:%s", r.Project.Docker.Image.Name, r.Project.Docker.Image.Tag)
+		return nil
+	}
+
+	if r.Project.Cypress.Version == "" {
+		return fmt.Errorf("no cypress version provided")
+	}
+	if r.Project.Cypress.Version == "latest" {
+		version, err := cypress.GetLatestDockerVersion()
+		if err != nil {
+			return err
+		}
+		log.Info().Msgf("Using Cypress %s", version)
+		r.Project.Cypress.Version = version
+	}
+
+	if r.Project.Docker.Image.Name == cypress.DefaultDockerImage && r.Project.Docker.Image.Tag == "" {
+		r.Project.Docker.Image.Tag = r.Project.Cypress.Version
+	}
+	if r.Project.Docker.Image.Name == "" {
+		r.Project.Docker.Image.Name = cypress.DefaultDockerImage
+		r.Project.Docker.Image.Tag = r.Project.Cypress.Version
+	}
+
+	dockerAvailability, cloudAvailability, err := cypress.IsCypressVersionAvailable(r.Project.Cypress.Version)
+	if err != nil {
+		msg := fmt.Sprintf("Unable to check Cypress availability: %s", err)
+		log.Error().Str("version", r.Project.Cypress.Version).Msg(msg)
+		return err
+	}
+	if !dockerAvailability {
+		msg := fmt.Sprintf("cypress %s is not available", r.Project.Cypress.Version)
+		log.Error().Str("version", r.Project.Cypress.Version).Msg(msg)
+		return errors.New(msg)
+	}
+	if !cloudAvailability {
```

Comment:
@FriggaHel what's the reason for checking cloud availability on docker mode? Is this method also called on "cypress sauce mode"?

Reply:
Good catch :) The cloud availability check shouldn't be performed for the docker mode.

Reply:
@vrunoa @alexplischke docker availability is checked just a couple of lines above. Since cloud availability info is provided at the same moment, this is used to display a warn (may be a lower level) to user stating that this version cannot be used on our cloud (and it has no overhead except this statement).

Reply:
Following that comment: https://github.com/saucelabs/saucectl/pull/158#discussion_r562024838
It will simply be removed.